pkg/http/middleware/analytics.go

```diff
@@ -0,0 +1,161 @@
+package middleware
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+	"regexp"
+
+	// TODO(remove): Remove auth packages
+	"go.signoz.io/signoz/pkg/query-service/auth"
+
+	"github.com/gorilla/mux"
+	v3 "go.signoz.io/signoz/pkg/query-service/model/v3"
+	"go.signoz.io/signoz/pkg/query-service/telemetry"
+	"go.uber.org/zap"
+)
+
+type AnalyticsMiddleware struct {
+	logger *zap.Logger
+}
+
+func NewAnalyticsMiddleware(logger *zap.Logger) *AnalyticsMiddleware {
```

Comment:
```suggestion
func NewAnalytics(logger *zap.Logger) *analytics {
```
Omit the middleware so that its more in line with other implementations in the package.

Reply:
thanks, have made the changes